Tests/UnitTests/Analytics/EventAnalyticsProviderTests.swift

```diff
@@ -0,0 +1,179 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+@_spi(AdyenInternal) @testable import Adyen
+@testable import AdyenNetworking
+
+final class EventAnalyticsProviderTests: XCTestCase {
+    
+    var eventDataSource = AnalyticsEventDataSource()
+
+    func testShouldNotSendEventsWhenNoAttemptId() {
+        let apiClient = APIClientMock()
+        let analyticsResponse = EmptyResponse()
+        let analyticsResult: Result<Response, Error> = .success(analyticsResponse)
+        
+        apiClient.mockedResults = [analyticsResult]
+        
+        let expectation = expectation(description: "should not be called")
+        expectation.isInverted = true
+        
+        apiClient.onExecute = { _ in
+            expectation.fulfill()
+        }
+        
+        let sut = createSUT(apiClient: apiClient)
+
+        sut.sendEventsIfNeeded()
+        
+        wait(for: [expectation], timeout: 0.1)
+    }
+
+    func testShouldNotSendEventWhenAttemptIdButNoEvents() {
```

Comment:
I am missing some verbs here :)
```suggestion
func testShouldNotSendEventWhenNoEventsAreStored()
```

The name also suggested the attempt id was set but I see no specific `checkoutAttemptId` setting in the preparation stage so I removed this from the test name.

Reply:
Good catch!

Reply:
I guess then all cases when you assume that no `checkoutAttemptId` is set are a bit obscure. I see there are two helpers: `createSUT()` and `createSUTWithSuccessMock()`
First one doesn't set any mocked response at all that might be misleading. Let's maybe add attemot id to the interface so it is super clear when you read the test?

For example:

func createSUT(checkoutAttemptId: String?) -> EventAnalyticsProvider {
}